tests/suites/helpers.function

```diff
@@ -107,15 +108,53 @@ typedef enum
  *
  * \param   TEST    The test expression to be tested.
  */
-#define TEST_ASSERT( TEST )                                 \
-    do {                                                    \
-       if( ! (TEST) )                                       \
-       {                                                    \
-          test_fail( #TEST, __LINE__, __FILE__ );           \
-          goto exit;                                        \
-       }                                                    \
+
+#define TEST_ASSERT( TEST )                                                   \
+    do {                                                                      \
+        if( ! (TEST) )                                                        \
+        {                                                                     \
+            set_test_result( #TEST, __LINE__, __FILE__, TEST_RESULT_FAILED ); \
```

Comment:
Why not keep using `test_fail()` and `test_skip()`?

Reply:
in order not to duplicate code.
I believe this came from [this comment](https://github.com/ARMmbed/mbedtls/pull/2124#discussion_r241462535)